internal/executor/build/task.go

```diff
@@ -38,7 +38,8 @@ type Task struct {
 
 func NewFromProto(protoTask *api.Task, logger logger.Lightweight) (*Task, error) {
 	// Create an instance that this task will run on
-	inst, err := instance.NewFromProto(protoTask.Instance, protoTask.Commands, logger)
+	inst, err := instance.NewFromProto(protoTask.Instance, protoTask.Commands, protoTask.Environment["CIRRUS_WORKING_DIR"],
```

Comment:
Should we expand things here for cases like `CIRRUS_WORKING_DIR: $SOME_DIR/foo`

Reply:
Good idea! Implemented in https://github.com/cirruslabs/cirrus-cli/pull/323/commits/a8154cc2526b7d83ecb8da56a2f64262483952f1.